lib/rubocop/cop/style/unpack_first.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for accessing the first element of `String#unpack`
+      # which can be replaced with the shorter method `unpack1`.
+      #
+      # @example
+      #
+      #   # bad
+      #   'foo'.unpack('h*').first
+      #   'foo'.unpack('h*')[0]
+      #   'foo'.unpack('h*').slice(0)
+      #   'foo'.unpack('h*').at(0)
+      #   'foo'.unpack('h*').take(1)
+      #
+      #   # good
+      #   'foo'.unpack1('h*')
+      #
+      class UnpackFirst < Cop
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.4
+
+        MSG = 'Use `String#unpack1(%<format>s)` instead of '\
```

Comment:
Why not `<%format>s.unpack1`? That reads better to me. Same for the other part of the message.

Reply:
Because `format` is [the argument](https://ruby-doc.org/core-2.4.3/String.html#method-i-unpack) to `unpack(1)`, not the string to be unpacked. 

I'm happy to leave the `String` prefix off, but I thought that was the suggestion you made in your [earlier comment](https://github.com/bbatsov/rubocop/pull/5643#discussion_r173700542).

Reply:
I meant I wanted the message to say something like `Use `foo.unpack1` instead of `foo...`. When using the actual code that you're checking in the messages they are easier to understand.

Reply:
Ok, I  updated the message to show this.